pytensor/tensor/subtensor.py

```diff
@@ -2682,10 +2689,68 @@ def grad(self, inputs, grads):
             rest
         )
 
+    @staticmethod
+    def non_contiguous_adv_indexing(node: Apply) -> bool:
```

Comment:
I put this as a staticmethod of the Op because I know we'll need to check for this case in many of the lifting rewrites